simulators/gaia/__init__.py

```diff
@@ -151,104 +157,107 @@ def _execute(self, msg):
             if args[1] not in range(1024):
                 return self._error(1010)
         cmd = getattr(self, cmd)
-        return cmd(args)
+        raw_response = cmd(args)
+        tail = ' %s%s' % (self.cmd_id, self.tail)
+        return self.header + str(raw_response) + tail
 
     def _error(self, error_code):
         error_string = self.errors.get(error_code)
-        return 'ERROR(%d)[%s](%s)%c' % (
+        return '%sERROR(%d)[%s](%s) %s%c' % (
+            self.header,
             error_code,
             error_string,
             error_string.encode('hex'),
+            self.cmd_id,
             self.tail
         )
 
-    def _idn(self, _):
-        return self.firmware_string + self.tail
+    def _idn(self, args):
+        return self.firmware_string
 
     def _setd(self, args):
         x = args[0]
         y = args[1]
         self.VD[x - 1] = y
-        return '%d%c' % (x, self.tail)
+        return x
 
     def _setg(self, args):
         x = args[0]
         y = args[1]
         self.VG[x - 1] = y
-        return '%d%c' % (x, self.tail)
+        return x
 
     def _setsg(self, args):
         x = args[0]
-        return '%d%c' % (x, self.tail)
+        return x
```

Comment:
Da qua in poi faccio una modifica: cancello l'assegnamento a x e restituisco direttamente `args[0]'.

Reply:
Si può anche variare da metodi a funzioni con `@staticmethod` e togliendo self

Reply:
Si, metto `@staticmethod` e poi aggiorno